synapse/api/auth.py

```diff
@@ -183,41 +182,29 @@ async def _wrapped_get_user_by_req(
 
             access_token = self.get_access_token_from_request(request)
 
-            (
-                user_id,
-                device_id,
-                app_service,
-            ) = await self._get_appservice_user_id_and_device_id(request)
-            if user_id and app_service:
+            # First check if it could be a request from an appservice
+            requester = await self._get_appservice_user(request)
+            if requester:
                 if ip_addr and self._track_appservice_user_ips:
                     await self.store.insert_client_ip(
-                        user_id=user_id,
+                        user_id=requester.user.to_string(),
                         access_token=access_token,
                         ip=ip_addr,
                         user_agent=user_agent,
-                        device_id="dummy-device"
-                        if device_id is None
-                        else device_id,  # stubbed
+                        device_id=requester.device_id,
```

Comment:
this looks like a change in behaviour? we will now set `device_id=None` where previously it was "dummy-device". Is that deliberate?

Reply:
It is, but probably not important: what it changes is that instead of inserting a `'dummy-device'` in the client IP tracking for appservices requests it inserts `NULL`

Reply:
> instead of inserting a 'dummy-device' in the client IP tracking for appservices requests it inserts NULL

right, but can you convince me that's an ok change to make?

Reply:
... and even if it is, do we really need to tie it into this refactoring PR?

Reply:
I reverted that and added a comment about it in f74ff1f64